src/comodo/jaxsimSimulator/jaxsimSimulator.py

```diff
@@ -0,0 +1,465 @@
+                        static_friction_coefficient=1.0,
+                    )
+                case _:
+                    raise ValueError(
+                        f"Invalid contact model type: {contact_model_type}"
+                    )
+
+        s = s or np.zeros(self._model.dofs())
+
+        self._data = js.data.JaxSimModelData.build(
+            model=self._model,
+            velocity_representation=VelRepr.Mixed,
+            base_position=jnp.array(xyz_rpy[:3]),
+            base_quaternion=jnp.array(self._RPY_to_quat(*xyz_rpy[3:])),
+            joint_positions=jnp.array(s),
+            contacts_params=contact_params,
+        )
+
+        if contact_model_type is not ContactModelEnum.VISCO_ELASTIC:
+            self._integrator = integrators.fixed_step.RungeKutta4.build(
+                dynamics=js.ode.wrap_system_dynamics_for_integration(
+                    model=self._model,
+                    data=self._data,
+                    system_dynamics=js.ode.system_dynamics,
+                )
+            )
+
+            self._integrator_state = self._integrator.init(
+                x0=self._data.state, t0=0, dt=self._dt
+            )
+
+        # TODO: expose these names as parameters
+        self._left_foot_link_idx = js.link.name_to_idx(
+            model=self._model, link_name=left_foot_link_name
+        )
+        self._right_foot_link_idx = js.link.name_to_idx(
+            model=self._model, link_name=right_foot_link_name
+        )
+        self._left_footsole_frame_idx = js.frame.name_to_idx(
+            model=self._model, frame_name=left_foot_sole_frame_name
+        )
+        self._right_footsole_frame_idx = js.frame.name_to_idx(
+            model=self._model, frame_name=right_foot_sole_frame_name
+        )
+
+        self._visualization_mode = visualization_mode
+
+        if self._visualization_mode is not None:
+            if self._visualization_mode not in ["record", "interactive"]:
+                raise ValueError(
+                    f"Invalid visualization mode: {self._visualization_mode}. "
+                    f"Valid options are: 'record', 'interactive'"
+                )
+
+            mjcf_string, assets = UrdfToMjcf.convert(
+                urdf=self._model.built_from,
+            )
+
+            self._mj_model_helper = MujocoModelHelper.build_from_xml(
+                mjcf_description=mjcf_string, assets=assets
+            )
+
+            self._recorder = MujocoVideoRecorder(
+                model=self._mj_model_helper.model,
+                data=self._mj_model_helper.data,
+                fps=30,
+            )
+
+        self._is_initialized = True
+
+    def set_input(self, input: npt.ArrayLike) -> None:
+        self._tau = jnp.array(input)
+
+    def step(self, n_step: int = 1) -> None:
+        assert (
+            self._is_initialized
+        ), "Simulator is not initialized, call load_model first."
+
+        for _ in range(n_step):
```

Comment:
We could think of managing this using a `jax.lax.scan` instead of a for-loop.

Reply:
I suggest postponing this for now since this loop has a bit of logic and conditionals inside, I think it isn't worth the effort.

Reply:
Yes, of course! It was just a comment for the future :)